platforms/android/src/org/crosswalk/engine/XWalkCordovaUiClient.java

```diff
@@ -172,12 +175,15 @@ public void onPageLoadStopped(XWalkView view, String url, LoadStatus status) {
     // File Chooser
     @Override
     public void openFileChooser(XWalkView view, final ValueCallback<Uri> uploadFile, String acceptType, String capture) {
-        uploadFile.onReceiveValue(null);
+        if (mFileChooser == null) {
+            mFileChooser = new XWalkFileChooser(parentEngine.cordova.getActivity());
```

Comment:
what's the differentiation between Activity and Application Context as as parameter to XWalkView.